Use fully qualified image name for envoy proxy image

vially opened the pull request:
This should fix an issue where `docker.io` might not be in the list of `unqualified-search-registries` when using Podman.

This _might_ be the root cause for #116.

BenTheElder commented:
Aside: you could coin a term "fully qualified image", but we are using the term FQDN incorrectly, would imply `docker.io.` (and therefore no DNS search), which we're not doing here (and shouldn't, I don't think it's valid in image names and would probably break things)

BenTheElder reviewed (commented):
/ok-to-test
/lgtm

thanks!

aojea commented:
/approve

vially commented:
> Aside: you could coin a term "fully qualified image", but we are using the term FQDN incorrectly, would imply `docker.io.` (and therefore no DNS search), which we're not doing here (and shouldn't, I don't think it's valid in image names and would probably break things)

Good point. I've updated the title and replaced FQDN with "fully qualified image name" which seems more accurate.

aojea commented:
/lgtm
